### ISSUE TITLE
Typeset plugin changes precedence of nav entry over h1

### ISSUE THREAD

**Issue by wilhelmer** [squidfunk/mkdocs-material#4795]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When the [typeset](https://squidfunk.github.io/mkdocs-material/reference/#built-in-typeset-plugin) plugin is enabled, a page's heading 1 takes precedence over the nav entry when determining the navigation title. Normally, this is vice versa.

This means that with the following configuration, the navigation title will be "My Topic Title" instead of "My Nav Title":

index.md:

```markdown
# My Topic Title

Lorem ipsum
```
 
mkdocs.yml:

```yml
nav:
  - My Nav Title: index.md
  - Topic: topic.md
...
plugins:
  - search
  - typeset
```

This may be desirable if you want to use Markdown formatting in the navigation, but I think there should be an option to override this behavior. For example, you may want nav titles that are shorter than the h1.

Also, I think it should be documented that if you want to apply Markdown formatting to the navigation, you actually have to do that via the h1, and not via `mkdocs.yml`.

Sample project: 
[typeset-sample.zip](https://github.com/squidfunk/mkdocs-material/files/10286855/typeset-sample.zip)

### Expected behaviour

See description.

### Actual behaviour

See description.

### Steps to reproduce

See description.

### Package versions

- Python: `3.11`
- MkDocs: `1.4.2`
- Material: latest Insiders version

### Configuration

```yaml
See description.
```

### System information

- Operating system: macOS
- Browser: FF

**Comment by normandy7** [comment]
Our desired behavior would also be for the `mkdocs.yml` nav title to take precedence in the left nav. What seems to happens on our end is that clicking an item in the left nav somewhat erratically updates that and all preceding nav items to the title set by h1 instead.

**Comment by squidfunk** [comment]
Thanks for reporting. Two problems are at play here. The problem with suceeding navigation items being not rendered correctly is fixed in [`d39b11ecd`](https://github.com/squidfunk/mkdocs-material-insiders/commit/d39b11ecd475de6d5a1f59e8c9c9044a074b1103), which was caused by an ordering issue among plugin events.

I'll shortly look into the original issue.

**Comment by squidfunk** [comment]
I _think_ I have a fix in [`1a04c7670`](https://github.com/squidfunk/mkdocs-material-insiders/commit/1a04c76705f8e18190ea125a07a46a48ba9f2bfe), which should cover all cases. The following table lists all sources of page title and their combinations. Thus, the built-in typeset plugin should now abide to the order imposed by MkDocs, [as described here](https://github.com/mkdocs/mkdocs/blob/56b235a8ad43f2300d17f87e6fa4de7a3d764397/mkdocs/structure/pages.py#L228-L257):

> Set the title for a Markdown document.
> Check these in order and use the first that returns a valid title:
>
> - value provided on init (passed in from config)
> - value of metadata 'title'
> - content of the first H1 in Markdown content
> - convert filename to title

---

If I didn't make a mistake, this will result in the following table of combinations:

| filename | config | meta | markdown | = navigation | = header |
| - | - | - | - | - | - |
| foo-bar.md | - | - | - | Foo Bar | Foo Bar |
| foo-bar.md | Foo | - | - | Foo | Foo |
| foo-bar.md | Foo | Bar | - | Foo | Bar |
| foo-bar.md | Foo | Bar | Baz | Foo | Bar |
| foo-bar.md | Foo | - | Baz | Foo | Foo |
| foo-bar.md | - | - | Baz | Baz | __Baz__ † |
| foo-bar.md | - | Bar | - | Bar | Bar |

† This is the only case where the plugin will override the navigation title, abiding to what is dictated by MkDocs.

---

As a first step, this should solve the issue at hand. In a second step, we could allow to change this behavior by adding a configuration setting to the plugin that allows to influence that order for the navigation. In a third step, we could allow to do the same for the header and footer, but that's just an idea.

**Comment by wilhelmer** [comment]
It's ultimately a 3-bit truth table, so there must be 8 entries. One is missing, just for the sake of completeness:

| filename   | config | meta | markdown | = navigation | = header |
| ---------- | ------ | ---- | -------- | ------------ | -------- |
| foo-bar.md | -      | Bar  | Baz      | Bar          | Bar      |

Anyway, thanks for fixing! 

Now for documentation, I'm not sure what to make of this. Since in 7 of 8 cases, the plugin will not alter the navigation title, maybe don't mention the navigation at all?

>The built-in typeset plugin preserves HTML formatting in ~the navigation and~ table of contents.

**Comment by squidfunk** [comment]
Well, I see this only as the start, since I think allowing to change precedence is something that would definitely make sense to add as a feature. First question on Twitter was if it's also possible to apply this for the header and footer titles, so I guess adding the possibility to specify this per page region makes sense.

And yes, I missed one, thanks for nothing! I'll include it in my comment 😊

**Comment by wilhelmer** [comment]
> thanks for nothing

How rude! One (character) too much this time 😆

**Comment by squidfunk** [comment]
Oh my god, I'm so sorry! That was not my intention 😅

**Comment by squidfunk** [comment]
Released as part of [9.0.3+insiders-4.27.1](https://github.com/squidfunk/mkdocs-material-insiders/releases/tag/9.0.3-insiders-4.27.1).
